src/functions.php

```diff
@@ -38,3 +40,32 @@ function usleep(int $time, LoopInterface $loop) : PromiseInterface
 {
     return Timer\resolve($time/1000000, $loop);
 }
+
+/**
+ * Returns the content type in MIME format, like text/plain or application/octet-stream, or FALSE on failure.
+ *
+ * @param string $fileName
+ * @param LoopInterface $loop
+ *
+ * @return PromiseInterface
+ */
+function mime_content_type(string $fileName, LoopInterface $loop) : PromiseInterface
+{
+    $process = new Process(sprintf('file %s --mime-type -b', $fileName));
+    $process->start($loop);
+    $deferred = new Deferred();
+    $data = '';
+    $stdout = $process->stdout;
+
+    $stdout->on('data', function($chunk) use (&$data) {
+        $data .= $chunk;
+    });
+
+    $stdout->on('end', function() use (&$data, $deferred) {
+        $deferred->resolve(trim($data));
+    });
+
```

Comment:
As @clue said, let's check for failures and in this case reject the promise with the reason of failure. Something like this:


```suggestion
    $process->on('exit', function($exitCode) use (&$data, $deferred) {
        if ($exitCode !== 0) {
            $deferred->reject(new \RuntimeException($data));
        }
    });
```

Reply:
Yes, working on that ATM: